### DISCUSSION TITLE
Autocomplete doesn't show Popover content if menuTrigger is set to 'mannual'

### DISCUSSION THREAD

**Discussion by Erichain** [heroui-inc/heroui#3431] [Q&A] [answered]
``` typescript
const [isOpen, setIsOpen] = useState(false);
const [keyword, setKeyword] = useState('');
const [items, setItems] = useState<{ id: string; name: string }[]>([]);

// ...

<SemiAutoComplete
  popoverProps={{
    isOpen,
  }}
  listboxProps={{
    emptyContent: 'Your own empty content text.',
  }}
  inputValue={keyword}
  placeholder="placeholder"
  aria-label="some text"
  items={items}
  onInputChange={(value) => {
    setKeyword(value);
    setIsOpen(true);
    setItems([
      {
        id: '1',
        name: '1',
      },
    ]);
  }}
  menuTrigger="manual"
   onOpenChange={(isOpen, menuTrigger) => {
     setIsOpen(isOpen);
     if (isOpen && menuTrigger === 'manual') {
       setKeyword(keyword);
       setItems(items);
     }
   }}
/>
```

So if I want to make Autocomplete a controlled component, where should I pass `isOpen`?

Now I pass this prop to `popoverProps`, but it doesn't work.

**Comment by wingkwong** [comment]
1. Please share the full reproducible code
2. Please share the version you use

**Reply by Erichain** [reply]
Sorry for delay.
I have written a reproducible project on Stackblitz, you can find it here: [nextui-autocomplete-problems](https://stackblitz.com/edit/nextui-autocomplete-problems?file=src%2FApp.tsx)

The version I use is the latest version: 2.4.2.

I encountered two problems(or bugs).

The first is, when I set `menuTrigger` to `focus` or `input`, the first time the popover shows, the input will lose focus unless the popover is always shown.

Another is when I set `menuTrigger` to `manual`, I just don't know how I can make it a controlled component, where should I pass the `isOpen` prop?

**Reply by wingkwong** [reply, marked as answer]
> The first is, when I set menuTrigger to focus or input, the first time the popover shows, the input will lose focus unless the popover is always shown.

I believe this issue has been resolved in current canary branch. I just tested with `0.0.0-canary-20240708000658`.

> Another is when I set menuTrigger to manual, I just don't know how I can make it a controlled component, where should I pass the isOpen prop?

I don't think it's related to `menuTrigger` at all. `menuTrigger` is the action that causes the menu to open. 

- input: menu opens when the user edits the input text.
- focus (default): menu opens when the user focuses the input.
- manual: menu only opens when the user presses the trigger button or uses the arrow keys.

And by the definition of `controlled`, it is about the value. If you are talking about programatically opening the autocomplete, then there is no way to do that.

For more about controlled autocomplete, please check out https://nextui.org/docs/components/autocomplete#controlled. 

> Not show popover if trigger method is set to manual

In your sandbox, I saw your second autocomplete actually works as expected. As mentioned above, it opens when you press the trigger button or use the arrow key.

**Reply by Erichain** [reply]
OK, got it. Thank you for your explanation. ❤️

> manual: menu only opens when the user presses the trigger button or uses the arrow keys.

Maybe I mistook this term for **programmatically control the open status for the popover**.

> I believe this issue has been resolved in current canary branch. I just tested with 0.0.0-canary-20240708000658.

So when will you release this version?

**Reply by wingkwong** [reply]
I don't have the exact date. Once we release, you will see the announcement in discord.

**Reply by Erichain** [reply]
Got it, thank you
